sources/packages/api/src/route-controllers/designation-agreement/designation-agreement.controller.ts

```diff
@@ -63,6 +65,17 @@ export class DesignationAgreementController {
         "Not able to create a designation agreement due to an invalid request.",
       );
     }
+
+    if (
+      await this.designationAgreementService.hasPendingDesignation(
```

Comment:
Just a suggestion. instead of making service call inside if, it can be assigned to a const and the const can be used in if statement.

Reply:
We can if we decided to go in this way. That was my first choice and I avoided because I though that there will a comment against it 😄 
I will create the const 👍

Reply:
Changed 👍